src/plugins/action_update_review_status/src/updateReviews.ts

```diff
@@ -0,0 +1,70 @@
+import databaseMethods from './databaseMethods'
+
+type ReviewStatus = 'Draft' | 'Submitted' | 'Changes Requested' | 'Pending' | 'Locked'
+interface Review {
+  reviewId: number
+  reviewAssignmentId: number
+  applicationId: number
+  reviewerId: number
+  level: number
+  reviewStatus: ReviewStatus
+}
+
+module.exports['updateReviews'] = async function (input: any, DBConnect: any) {
+  const db = databaseMethods(DBConnect)
+
+  console.log('Updating reviews status...')
+
+  const { applicationId, changedApplicationResponses = [] } = input
+
+  const reviewsToUpdate = []
+
+  console.log('changedApplicationResponses', changedApplicationResponses)
+
+  try {
+    // Get reviews/review assignments (with status) matching application_id
+    const reviews = (await db.getAssociatedReviews(applicationId)).filter((review: Review) =>
+      ['Submitted', 'Locked', 'Draft'].includes(review.reviewStatus)
+    )
+    // Deduce which ones should be updated
+    for (const review of reviews) {
+      const { reviewAssignmentId, level, reviewStatus } = review
+      if (level > 1) reviewsToUpdate.push({ ...review, reviewStatus: 'Pending' })
+      else if (await haveAssignedResponsesChanged(reviewAssignmentId))
+        reviewsToUpdate.push({ ...review, reviewStatus: 'Pending' })
+      else if (reviewStatus === 'Pending' || reviewStatus === 'Locked')
+        reviewsToUpdate.push({ ...review })
```

Comment:
should this have reviewStatus ?

also reviewStatus === 'Pending' would be filtered out (so not sure if you want to have it here)

Reply:
> should this have reviewStatus ?

Yeah, if we want "Locked" ones to now become "Pending". Have updated now 0bf1dc0.

> also reviewStatus === 'Pending' would be filtered out (so not sure if you want to have it here)

Yeah, that's true. The thought was that "Pending" reviews should have their status_history timestamp updated anyway, but I guess that doesn't need to happen does it? 

Have now removed, already "Pending" review will be untouched. 37c8b3e